chesster.js

```diff
@@ -27,177 +30,161 @@ var BOARD_5_RATING = 16;
-            var found = false;
-            commands.forEach(function(c) {
-                if (found) { return; }
-                if (levenshtein.get(c, item) < c.length / 2) {
-                    found = true;
-                    command = item;
-                }
-            });
-            if (!found) {
-                target = item;
-            }
-        });
-    }
-    command = fuzzy_match(command, commands);
-    target = fuzzy_match(target, LONEWOLF_TARGETS.concat(TEAM_TARGETS));
-    if (command.length > 1) {
-        console.log("Ambiguous command: ", command, target);
-        return {
-            command: null,
-            target: null
-        };
-    }
-    command = command[0];
-    var team_votes = 0;
-    var wolf_votes = 0;
-    target.forEach(function(item) {
-        var is_team = TEAM_TARGETS.indexOf(item) > -1;
-        var is_lonewolf = LONEWOLF_TARGETS.indexOf(item) > -1;
-        if (is_team) team_votes++;
-        if (is_lonewolf) wolf_votes++;
 
-    });
-    var retval = {
-        command: command,
-        target: null,
-    };
-    if (team_votes == wolf_votes) {
-        console.log("Ambiguous target!");
-    } else if (team_votes > wolf_votes) {
-        retval.target = "team";
-    } else if (wolf_votes > team_votes) {
-        retval.target = "lonewolf";
-    }
-    return retval;
-}
-
-controller.spawn({
-  token: process.env.token
-}).startRTM(function(err, bot) {
+function update_users(bot){
+    users.list = [];
     // @ https://api.slack.com/methods/users.list
     bot.api.users.list({}, function (err, response) {
+        if (err) {
+            throw new Error(err);
+        }
+
         if (response.hasOwnProperty('members') && response.ok) {
             var total = response.members.length;
-            for (var i = 0; i < total; i++) {
+             for (var i = 0; i < total; i++) {
                 var member = response.members[i];
-                users[member.name] = member;
+                users.list[member.name] = member;
             }
         }
         console.log("info: got users");
     });
+}
 
+function update_channels(bot){
+    channels.list = [];
+    channels.byId = [];
     // @ https://api.slack.com/methods/channels.list
     bot.api.channels.list({}, function (err, response) {
+        if (err) {
+            throw new Error(err);
+        }
+
         if (response.hasOwnProperty('channels') && response.ok) {
             var total = response.channels.length;
             for (var i = 0; i < total; i++) {
                 var channel = response.channels[i];
-                channels[channel.name] = channel;
-                channelsByID[channel.id] = channel;
+                channels.list[channel.name] = channel;
+                 channels.byId[channel.id] = channel;
             }
         }
         console.log("info: got channels");
     });
+
+}
+
+/* 
+   updates the user list
+   updates the channel lists
+   then repeats in new threat
```

Comment:
Ha. Must have been playing chess when I wrote that
On Apr 10, 2016 11:44 AM, lakinwecker <notifications@github.com> wrote:In chesster.js:
>              }
>          }
>          console.log("info: got channels");
>      });
> +
> +}
> +
> +/* 
> +   updates the user list
> +   updates the channel lists
> +   then repeats in new threat

threat - > thread

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub